deepchem/feat/tests/test_mxmnet_featurizer.py

```diff
@@ -0,0 +1,129 @@
+
+        assert (graph_feat[0].edge_index == self.edge_index[0]).all()
+
+    def test_featurizer_general_case(self):
+        """
+        Test for featurization of "CC(=O)C" using `MXMNetFeaturizer` class.
+        """
+
+        featurizer = dc.feat.molecule_featurizers.mxmnet_featurizer.MXMNetFeaturizer(
+        )
+        graph_feat = featurizer.featurize(self.smiles)
+        assert len(graph_feat) == 4
+
+        assert graph_feat[1].num_nodes == 4
+        assert graph_feat[1].num_node_features == 1
+        assert graph_feat[1].node_features.shape == (4, 1)
+        assert graph_feat[1].num_edges == 6
+        assert (graph_feat[1].node_features == self.node_features[1]).all()
+
+        assert (graph_feat[1].edge_index == self.edge_index[1]).all()
+
+    def test_featurizer_single_atom(self):
+        """
+        Test for featurization of "C" using `MXMNetFeaturizer` class.
+        """
+
+        featurizer = dc.feat.molecule_featurizers.mxmnet_featurizer.MXMNetFeaturizer(
+        )
+        graph_feat = featurizer.featurize(self.smiles)
+        assert len(graph_feat) == 4
+
+        assert graph_feat[2].num_nodes == 1
+        assert graph_feat[2].num_node_features == 1
+        assert graph_feat[2].node_features.shape == (1, 1)
+        assert graph_feat[2].num_edges == 0
+        assert (graph_feat[2].node_features == self.node_features[2]).all()
+
+        assert (graph_feat[2].edge_index == self.edge_index[2]).all()
+
+    def test_featurizer_other_atom(self):
+        """
+        Test for featurization of "CP" using `MXMNetFeaturizer` class.
+        Since the smile contains P which is not supported by featurizer, the featurization process terminates and the featurizer returns an empty numpy array.
+        """
+
+        featurizer = dc.feat.molecule_featurizers.mxmnet_featurizer.MXMNetFeaturizer(
+        )
+        graph_feat = featurizer.featurize(self.smiles)
+        assert len(graph_feat) == 4
+
+        assert graph_feat[3].shape == (0,)
+
+    def test_node_pos_features(self):
+        """
+        Test for featurization of "C" using `MXMNetFeaturizer` class.
+        It checks whether node_pos_features are handled properly.
+        """
+        smile = ['C']
+        pos_x1 = [np.array([-0.0127, 0.0022, 1.0117, -0.5408, -0.5238])]
+        pos_y1 = [np.array([1.0858, -0.0060, 1.4638, 1.4475, 1.4379])]
+        pos_z1 = [np.array([0.0080, 0.0020, 0.0003, -0.8766, 0.9064])]
+
+        featurizer = dc.feat.molecule_featurizers.mxmnet_featurizer.MXMNetFeaturizer(
+            is_adding_hs=True)
+        graph_feat = featurizer.featurize(smile,
+                                          pos_x=pos_x1,
+                                          pos_y=pos_y1,
+                                          pos_z=pos_z1)
+        assert type(graph_feat[0].node_pos_features) == np.ndarray
+        assert (graph_feat[0].node_pos_features == node_pos).all
```

Comment:
To avoid floating point errors, it is better to use something like this for assertion of float values: `numpy.allclose(output, required_output, atol=1e-3)`

Reply:
done